aosp_diff/celadon_ivi/frameworks/av/cmds/stagefright/0002-Enable-stagefright-and-recorder-for-debug.patch

```diff
@@ -0,0 +1,528 @@
+-        int16_t amplitude = (int16_t)(32767.0 * sin(x));
+-        
+-        *ptr++ = amplitude;
+-        if (mNumChannels == 2) {
+-            *ptr++ = amplitude;
++        if (mFormat == AUDIO_FORMAT_PCM_16_BIT) {
++            int16_t amplitude = (int16_t)(32767.0 * sin(x));
++
++            for (int32_t j = 0; j < mNumChannels; j++) {
++                *((int16_t *)ptr) = amplitude;
++
++                ptr += audio_bytes_per_sample(mFormat);
++            }
++        } else if (mFormat == AUDIO_FORMAT_PCM_24_BIT_PACKED) {
++            int32_t amplitude = (int32_t)(8388607.0 * sin(x));
++
++            for (int32_t j = 0; j < mNumChannels; j++) {
++                *((int8_t *)ptr)       = *((int8_t *)(&amplitude));
++                *((int8_t *)(ptr + 1)) = *((int8_t *)(&amplitude) + 1);
++                *((int8_t *)(ptr + 2)) = *((int8_t *)(&amplitude) + 2);
++
++                ptr += audio_bytes_per_sample(mFormat);
++            }
++        } else if (mFormat == AUDIO_FORMAT_PCM_8_24_BIT) {
++            int32_t amplitude = (int32_t)(8388607.0 * sin(x));
++
++            for (int32_t j = 0; j < mNumChannels; j++) {
++                *((int8_t *)ptr)       = *((int8_t *)(&amplitude));
++                *((int8_t *)(ptr + 1)) = *((int8_t *)(&amplitude) + 1);
++                *((int8_t *)(ptr + 2)) = *((int8_t *)(&amplitude) + 2);
++                *((int8_t *)(ptr + 3)) = 0;
++
++                ptr += audio_bytes_per_sample(mFormat);
++            }
++        } else if (mFormat == AUDIO_FORMAT_PCM_32_BIT) {
++            int32_t amplitude = (int32_t)(2147483647.0 * sin(x));
++
++            for (int32_t j = 0; j < mNumChannels; j++) {
++                *((int32_t *)ptr) = amplitude;
++
++                ptr += audio_bytes_per_sample(mFormat);
++            }
+         }
+ 
+         x += k;
+diff --git a/cmds/stagefright/SineSource.h b/cmds/stagefright/SineSource.h
+index 6f1d98c769..2ed1e335b3 100644
+--- a/cmds/stagefright/SineSource.h
++++ b/cmds/stagefright/SineSource.h
+@@ -4,13 +4,14 @@
+ 
+ #include <media/stagefright/MediaSource.h>
+ #include <utils/Compat.h>
++#include <system/audio.h>
+ 
+ namespace android {
+ 
+ class MediaBufferGroup;
+ 
+ struct SineSource : public MediaSource {
+-    SineSource(int32_t sampleRate, int32_t numChannels);
++    SineSource(int32_t sampleRate, int32_t numChannels, audio_format_t format = AUDIO_FORMAT_PCM_16_BIT);
+ 
+     virtual status_t start(MetaData *params);
+     virtual status_t stop();
+@@ -30,6 +31,7 @@ private:
+     bool mStarted;
+     int32_t mSampleRate;
+     int32_t mNumChannels;
++    audio_format_t mFormat;
+     size_t mPhase;
+ 
+     MediaBufferGroup *mGroup;
+diff --git a/cmds/stagefright/WavRecorder.cpp b/cmds/stagefright/WavRecorder.cpp
+new file mode 100644
+index 0000000000..7ab3b5ab9a
+--- /dev/null
++++ b/cmds/stagefright/WavRecorder.cpp
+@@ -0,0 +1,219 @@
++/*
++ * Copyright (C) 2014 The Android Open Source Project
```

Comment:
If this file is written by us, please change the copyright to Intel

Reply:
updated